### PR TITLE
featL added edit profile,follow/unfollow features

### PR DESCRIPTION
**Pull request by Chakravarthi0**
now users can:
 - change their profile picture
 - change bio and url
 - follow/unfollow other users

### CODE REVIEW

File: src/features/user/helpers/followUser.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    try {
+      const { data } = await axios.post(
+        `/api/users/follow/${idToFollow}`,
+        {},
+        { headers: { authorization: token } }
+      );
+      toast.success("Followed user!");
+      return data;
+    } catch (err) {
+      console.log(err);
```
Comment (pritam-kr) [inline src/features/user/helpers/followUser.js:17]: remove the consoles

**Review by pritam-kr — APPROVED** [review]
You have written good clean code.
good folder sructure.
 suggestion: 
 - the width of the post button on the feed page can be increase ??
 - toggle theme button looking more CTA to the user .

Other than everything is good to merge.

File: src/components/edit-profile-modal/EditProfileModal.jsx [hunk 2]
```diff
@@ -1,40 +1,109 @@
...
+                    ? URL.createObjectURL(imageUrl)
+                    : currentUser?.profileUrl
+                }
                 alt="user image"
               />
-              <div className="absolute right-0 bottom-0 flex justify-center items-center h-9 w-9 bg-black  text-white dark:bg-white dark:text-black rounded-full">
+              {/* <div className="absolute right-0 bottom-0 flex justify-center items-center h-9 w-9 bg-black  text-white dark:bg-white dark:text-black rounded-full">
                 <FiCamera className="text-xl cursor-pointer" />
-              </div>
+              </div> */}
```
Comment (gopalbbharadva) [inline src/components/edit-profile-modal/EditProfileModal.jsx]: You can remove these comments if not needed.

File: src/components/post/Post.jsx [hunk 3]
```diff
@@ -1,42 +1,48 @@
...
 
 const Post = () => {
-  const { toggleShowPostModal } = usePostModal();
   const [showPostOptions, setShowPostOptions] = useState(false);
+  const optionsModalRef = useRef(null);
+  const dispatch = useDispatch();
   const navigate = useNavigate();
   const toggleShowOptions = () => {
     setShowPostOptions((prev) => !prev);
   };
```
Comment (gopalbbharadva) [inline src/components/post/Post.jsx:16]: it can write in single line.

File: src/features/user/helpers/getUsers.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { createAsyncThunk } from "@reduxjs/toolkit";
+import axios from "axios";
+
+const getUsers = createAsyncThunk("user/getUsers", async (usrInput) => {
+  try {
+    const { data } = await axios.get("/api/users");
+    return data;
+  } catch (err) {
+    console.log(err);
```
Comment (gopalbbharadva) [inline src/features/user/helpers/getUsers.js:9]: you can remove this console

**Review by gopalbbharadva — APPROVED** [review]
Feature working fine, edit use details also working fine.
Bug: Updated user profile image in not sowing in posts avatar of that user.

**Review by manojasarsa — APPROVED** [review]
The code is pretty well structured.
Also, the functionality is working as expected.
Good work!

**Review by msdeshmukh009 — APPROVED** [review]
**Feature is working as expected.**

**Suggestions**: 
- You can close the followers/following modal after the user gets redirected to any user's profile. 
- I think the avatar on the profile page has a fixed height, it's better to use relative units there otherwise it will overflow.

**Comment by Chakravarthi0** [comment]
> **Feature is working as expected.**
> 
> **Suggestions**:
> 
> * You can close the followers/following modal after the user gets redirected to any user's profile.
> * I think the avatar on the profile page has a fixed height, it's better to use relative units there otherwise it will overflow.

fixed, thanks for reviewing

**Comment by Chakravarthi0** [comment]
> Feature working fine, edit use details also working fine. Bug: Updated user profile image in not sowing in posts avatar of that user.

It takes a while for the image to get uploaded, so, for a bried moment, the profile image will be empty. Added a loader to indicate that image is being uploaded, thanks for reviewing

**Comment by Chakravarthi0** [comment]
> You have written good clean code. good folder sructure. suggestion:
> 
> * the width of the post button on the feed page can be increase ??
> * toggle theme button looking more CTA to the user .
> 
> Other than everything is good to merge.

fixed, thanks for reviewing

**Pull request merged**